crates/bevy_render/src/texture/icon.rs

```diff
@@ -0,0 +1,76 @@
+use crate::texture::Image;
+use bevy_asset::{Assets, Handle};
+use bevy_ecs::{
+    prelude::{Component, Entity, NonSendMut, Query, Res},
+    system::Commands,
+};
+use bevy_log::{error, info};
+use bevy_winit::WinitWindows;
+use winit::window::Icon;
+
+/// An icon that can be placed at the top left of the window.
+#[derive(Component, Debug)]
+pub struct WindowIcon(pub Option<Handle<Image>>);
```

Comment:
A simple `new(handle: Handle<Image>)` method would go a long way to improving usability here :)